src/zkevm_specs/evm/execution/calldatasize.py

```diff
@@ -0,0 +1,26 @@
+from ..instruction import Instruction, Transition
+from ..table import CallContextFieldTag
+from ..opcode import Opcode
+from ...util.param import N_BYTES_U64
+
+
+def calldatasize(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+    instruction.constrain_equal(opcode, Opcode.CALLDATASIZE)
+
+    # check [rw_table, call_context] table for call data length and compare
+    # against stack top after push.
+    instruction.constrain_equal(
+        instruction.int_to_rlc(
+            instruction.call_context_lookup(CallContextFieldTag.CallDataLength),
+            N_BYTES_U64,
```

Comment:
> Recently I am thinking maybe we should have a 32 bytes (full word) base conversion for such simple gadget (no range check, only base conversion), since the worst case already has such "full word base conversion", and in the future we might refactor them to share the same constraint. So perhaps for simplicity, we just do full word decomposition.

So if I understand it correctly, since other opcodes already to the RLC with 32 bytes, doing it here as well doesn't introduce any extra cost right?  So should I replace the `N_BYTES_U64` by `N_BYTES_WORD`?

Also, should I apply this same change to `caller.py`? which now is:
```py
    # check [rw_table, call_context] table for caller address and compare with
    # stack top after push
    instruction.constrain_equal(
        instruction.int_to_rlc(
            instruction.call_context_lookup(CallContextFieldTag.CallerAddress),
            N_BYTES_ACCOUNT_ADDRESS,
        ),
        instruction.stack_push(),
    )
```

Reply:
> So if I understand it correctly, since other opcodes already to the RLC with 32 bytes, doing it here as well doesn't introduce any extra cost right?

Yes, ideally we only need to sum up their selector for such constraints. Tho such optimization has not been implemented in circuit.

> So should I replace the `N_BYTES_U64` by `N_BYTES_WORD`? Also, should I apply this same change to caller.py?

If we are preparing for the optimization above (to reuse identical constraint as much as possible), I think so. But perhaps we can add a note that, it can do base conversion with 32 bytes is becuase it's range is already checked in the beginning and it's also a read-only data, then such extra 12/24 bytes unchecked will not break things.

Reply:
I've created this issue https://github.com/appliedzkp/zkevm-specs/issues/101 and added a note in the code so that we can think about it.